receiver/sqlqueryreceiver/integration_test.go

```diff
@@ -241,6 +241,128 @@ func TestOracleDBIntegration(t *testing.T) {
 	testMovieMetrics(t, rms.At(0), genreKey)
 }
 
+func TestMysqlIntegration(t *testing.T) {
+	externalPort := "13306"
+	internalPort := "3306"
+	waitStrategy := wait.ForListeningPort(nat.Port(internalPort)).WithStartupTimeout(2 * time.Minute)
```

Comment:
Are we _really_ looking to wait up to 2 minutes of CI time for the port to become available? Can we speed this up a bit? How long would we typically wait?

Reply:
2 minutes is the maximum time we will wait for the container to be available. Usually, it takes a little less time. On my machine, it took less than 1 minute.

The default value of this parameter is one minute. However, it is increased in all other integration tests for this module: to [2 minutes for PostgreSQL](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/sqlqueryreceiver/integration_test.go#L40) and to [15 minutes for Oracle](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/sqlqueryreceiver/integration_test.go#L173). I decided to set it to 2 minutes (like for PosgreSQL), because on my laptop the container for MySQL takes a little longer to start than the one for PostgreSQL.

Of course, after merging, I will observe whether the test is performed correctly and if there is a need to adjust this time, I will do so

Reply:
If this takes about a minute to be ready, we might want to split this into a different step/suite so that it doesn't block the entire test suite for a whole minute. Our CI is already too slow for my taste, and we can't afford to have components add slow operations.

Reply:
This test is part of the integration test suite. In my opinion, this is the test suite where we should put the slower ones. 

Running the test for MySQL requires less than a minute on our CI. [It took 158.981s to run all 3 integration tests in the sqlqueryreceiver module](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4637478361/jobs/8244496982?pr=20744#step:5:226). [Whole integration tests run in just 16m 27s](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4637478361/jobs/8244496982?pr=20744). To me, it doesn't look like too much execution time for such a large project.

Just like @astencel-sumo , I believe that such a component should have integration tests for each supported database